core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -702,7 +715,7 @@ public Index(Term uid, ParsedDocument doc) {
         }
 
         public Index(Term uid, ParsedDocument doc, long version) {
-            this(uid, doc, version, VersionType.INTERNAL, Origin.PRIMARY, System.nanoTime());
+            this(uid, doc, -1, version, VersionType.INTERNAL, Origin.PRIMARY, System.nanoTime());
```

Comment:
Here's another place to maybe use a [field](https://github.com/elastic/elasticsearch/pull/14651/files#r45225330).